packages/block-library/src/query-pagination/index.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ * Server-side rendering of the `core/query-pagination` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/query-pagination` block on the server.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string Returns the pagination for the query.
+ */
+function render_block_core_query_pagination( $attributes, $content, $block ) {
+	$page_key = 'query-' . $block->context['id'] . '-page';
+	$page     = empty( $_GET[ $page_key ] ) ? 1 : $_GET[ $page_key ];
```

Comment:
We should probably be sanitizing these inputs, as they would be very easy to manipulate in ways we might forget to anticipate in the future (i.e. injection).

e.g.

```suggestion
	$page     = empty( $_GET[ $page_key ] ) ? 1 : filter_var( $_GET[ $page_key ], FILTER_VALIDATE_INTEGER );
```

https://www.php.net/manual/en/function.filter-var.php

To a more general point, typically it should be a red flag to need to be manually validating values like this. Which could be a sign of how we're reaching beyond the block's own attributes and context to query parameters of the URL. I don't know that there's much of an alternative, but it could give us some headaches in the future, or at least need to be especially cautious about. It's partly why functions like [`get_query_var`](https://developer.wordpress.org/reference/functions/get_query_var/) exist, even if something akin to a query argument getter, providing value by preventing people from shooting themselves in the foot.

Reply:
I updated it.

Yeah, we should explore having our own safe utilities for this.